open-commerce-search-api/src/main/java/de/cxp/ocs/api/searcher/Searcher.java

```diff
@@ -19,6 +20,6 @@
 	 * @return
 	 * @throws IOException
 	 */
-	public SearchResult find(String tenant, String searchQuery, SearchParams parameters) throws IOException;
+	public SearchResult find(String tenant, Query query, SearchParams parameters) throws IOException;
```

Comment:
@pavelpenchev I don't see any advantage of this separation anymore. I'd suggest to put all parameters into a single object - event a `Map<String, String>` or `URI` would be ok for me: All the parsing and validation has to happen on service end anyways.